src/main/java/com/google/roomies/database/NoSQLDatabase.java

```diff
@@ -26,21 +28,29 @@
   */  
   public void addListingAsMap(Listing listing);
 
+  /**
+  * Add a comment to a given listing.
+  *
+  * @param comment a Comment instance
+  * @param listingId ID of the listing in database
+  */
+  public void addCommentAsMapToListing(Comment comment, String listingId) throws 
```

Comment:
I don't think the "AsMap" is relevant here, since you're passing in the value object? Seems like an implementation detail you can hide and get a cleaner method name.

Reply:
Done